File: src/main/java/com/ihaproject/model/Telemetry.java
package com.ihaproject.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
public class Telemetry {

   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   private Long ihaId;

   private Double startLatitude;
   private Double startLongitude;
   private double currentLatitude;
   private double currentLongitude;
   private Double targetLatitude;
   private Double targetLongitude;
   private Double missionProgress;
   private int altitude;
   private double speed;
   private int battery;

   private LocalDateTime timestamp;

   @PrePersist
   public void prePersist() {
      this.timestamp = LocalDateTime.now();
   }
}

File: frontend/src/types/Telemetry.ts
export interface Telemetry {
   ihaId: number;
   startLatitude: number | null;
   startLongitude: number | null;
   currentLatitude: number;
   currentLongitude: number;
   targetLatitude: number | null;
   targetLongitude: number | null;
   altitude: number;
   speed: number;
   battery: number;
   missionProgress: number | null;
   timestamp: string;
}

